### PR TITLE
V3.4.8

### PR DESCRIPTION
**Pull request by spgreenberg**
## Changes proposed in this pull request:

This is a more aggressive edit to update and remove unused code. It is unlikely we will need to pull updates from the upstream.

- Updates the open-saml dependencies
- Changes logging to logback as this is what shibboleth uses. I have been unable to get logs out of this plugin in our bosh release and I hope this fixes that.
- Remove unused code
- Removes unused imports
- VS Code also did some formatting. None of the functionality was changed.

## security considerations

- Removing unused code is good.
- Updating the latest saml dependencies
- Logs are necessary for visibility

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -28,18 +28,14 @@ subprojects {
     implementation group: 'org.codehaus.janino', name: 'janino', version:'2.7.8'
     implementation group: 'net.shibboleth.utilities', name: 'java-support', version:'7.2.0'
     implementation group: 'javax.mail', name: 'mail', version:'1.4.7'
-    implementation group: 'org.opensaml', name: 'opensaml-messaging-api', version:'3.2.0'
+    implementation group: 'org.opensaml', name: 'opensaml-messaging-api', version:'3.4.6'
     implementation group: 'org.springframework', name: 'spring-context-support', version:'4.2.3.RELEASE'
     implementation group: 'org.springframework', name: 'spring-jdbc', version: '4.2.3.RELEASE'
-    implementation group: 'org.slf4j', name: 'slf4j-api', version:'1.7.12'
+    implementation 'ch.qos.logback:logback-classic:1.2.3'
```
Comment (spgreenberg) [inline build.gradle:34]: Logback contains slf4j.

File: build.gradle [hunk 2]
```diff
@@ -28,18 +28,14 @@ subprojects {
...
     implementation 'javax.servlet:javax.servlet-api:3.0.1'
     testImplementation group: 'org.springframework', name: 'spring-test', version:'4.2.3.RELEASE'
     testImplementation group: 'net.shibboleth.utilities', name: 'java-support', version:'7.2.0'
-    testImplementation group: 'ch.qos.logback', name: 'logback-classic', version:'1.1.3'
```
Comment (spgreenberg) [inline build.gradle:38]: Dependencies in 'implementation' are availabe in 'testImplementation'

File: totpauth-impl/src/main/java/net/kvak/shibboleth/totpauth/authn/impl/CheckForSeed.java [hunk 3]
```diff
@@ -2,24 +2,22 @@
...
+import net.kvak.shibboleth.totpauth.api.authn.SeedFetcher;
 import net.kvak.shibboleth.totpauth.api.authn.context.TokenUserContext;
+import net.kvak.shibboleth.totpauth.api.authn.context.TokenUserContext.AuthState;
+import net.shibboleth.idp.authn.AuthnEventIds;
 import net.shibboleth.idp.authn.context.AuthenticationContext;
 import net.shibboleth.idp.authn.context.UsernamePasswordContext;
 import net.shibboleth.idp.profile.AbstractProfileAction;
 import net.shibboleth.utilities.java.support.annotation.constraint.NotEmpty;
 import net.shibboleth.utilities.java.support.component.ComponentInitializationException;
-import net.shibboleth.utilities.java.support.primitive.StringSupport;
```
Comment (spgreenberg) [inline totpauth-impl/src/main/java/net/kvak/shibboleth/totpauth/authn/impl/CheckForSeed.java:16]: Unused imports

File: totpauth-impl/src/main/java/net/kvak/shibboleth/totpauth/authn/impl/CheckForSeed.java [hunk 4]
```diff
@@ -41,10 +38,10 @@ public void setseedFetcher(@Nonnull @NotEmpty final SeedFetcher seedFetcher) {
 		this.seedFetcher = seedFetcher;
 	}
 
-    @Override
-    protected void doInitialize() throws ComponentInitializationException {
-        super.doInitialize();
-    }
+	@Override
```
Comment (spgreenberg) [inline totpauth-impl/src/main/java/net/kvak/shibboleth/totpauth/authn/impl/CheckForSeed.java:41]: This is just vs code formatting

**Review by mheadd — APPROVED** [review]
LGTM! Thanks for adding those comments explaining the changes. Super helpful.

**Review by soutenniza — APPROVED** [review]
Thanks, Steve!

**Pull request merged**